### ISSUE TITLE
Make ArangoCollection#save behave consistently

### ISSUE THREAD

**Issue by pluma** [arangodb/arangodb#1733]
It can be unintuitive that ArangoCollection#save/insert requires different arguments for document and edge collections. I would propose adjusting the edge version so that it can optionally take the same arguments as the document version as long as the data object contains the `_from` and `_to` properties. This would behave consistently with how `_key` is specified.

The main advantage would be that whether a document to create represents a vertex or edge would no longer affect the API that needs to be invoked in order to create it. ArangoJS already supports this for this reason (though the to/from parameters are specified after the data for convenience): https://github.com/arangodb/arangojs#edgecollectionsave

This is pretty straightforward for the client/arangosh implementation in JS:

```diff
diff --git a/js/client/modules/@arangodb/arango-collection.js b/js/client/modules/@arangodb/arango-collection.js
index eb9dda1..469a0a2 100644
--- a/js/client/modules/@arangodb/arango-collection.js
+++ b/js/client/modules/@arangodb/arango-collection.js
@@ -962,11 +962,18 @@ ArangoCollection.prototype.insert = function (from, to, data, waitForSync) {
     url = "/_api/document?collection=" + encodeURIComponent(this.name());
   }
   else if (type === ArangoCollection.TYPE_EDGE) {
+    if (!data && from && typeof from === 'object' && from.hasOwnProperty("_from") && from.hasOwnProperty("_to")) {
+      data = from;
+      waitForSync = to;
+      from = data._from;
+      to = data._to;
+    }
+
```

The server/arangod version however seems to be implemented natively. My C++-Fu is not strong enough to patch that one.

**Comment by neunhoef** [comment]
I think this is a good idea, but might be more work on the C++ side.

**Comment by pluma** [comment]
@neunhoef it would be trivial to just change the JS API within JS. But the save/insert methods seem to be directly implemented in C++. Would it be possible to rename the native methods (e.g. `_insert`) and then just provide a thin JS wrapper that does the argument checking as in arangosh? We already have wrapper functions like this for most internal APIs.

**Comment by neunhoef** [comment]
This will be addressed more thoroughly in 3.0, for which we do a full revision of the document API in JS and HTTP.

**Comment by neunhoef** [comment]
Done in 3.0
